sklearn/preprocessing/data.py

```diff
@@ -1531,6 +1531,24 @@ def _fit_transform(self, X):
             out = out[:, active_features]
             self.active_features_ = active_features
 
+            fi = self.feature_indices_
+
+            # Find indexes feature indices aprropriately
+            split_indices = np.searchsorted(active_features,
```

Comment:
gotta run for now but it is not obvious what this line does (maybe I just need to look at it for longer). But "appropriately" ins not very informative ;)